app/controllers/journals_controller.rb

```diff
@@ -0,0 +1,68 @@
+class JournalsController < ApplicationController
+  include MakeNotice
+
+  before_action :set_journal, only: [:show, :edit, :update, :destroy]
+  before_action :set_journaled
+
+  def index
+    @journals = Journal.where(journal_relations.except(:user_id))
+  end
+
+  def show; end
+
+  def new
+    @journal = Journal.new(journal_relations)
+    authorize(@journal)
+  end
+
+  def edit; end
+
+  def create
+    @journal = Journal.new(journal_params.merge(journal_relations))
+    authorize(@journal)
+    if @journal.save
+      redirect_to @journaled, make_notice
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @journal.update(journal_params)
+      redirect_to @journaled, make_notice
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @journal.destroy
+    redirect_to @journaled, make_notice
+  end
+
+  private
+
+  def journal_relations
+    {
+      journalable_type: @journaled.class.model_name.name,
+      journalable_id: @journaled.id,
+      user_id: current_user.id
+    }
+  end
+
+  def set_journal
+    @journal = Journal.find(params[:id].to_i)
```

Comment:
To integer transformation is not necessary, Rails is smart enough - so `Journal.find("1")` is the same as `Journal.find(1)`. Try it out in console, it will work. Will remove it before merging it.